features/deduper.js

```diff
@@ -0,0 +1,61 @@
+const deduper = {
+  monitoredChannels: [
+    "102860784329052160", // #general
+    "103696749012467712", // #need-help-0
+    "565213527673929729" // #need-help-1
+  ],
+
+  dupeExpireTime: 2 * 60 * 1000, // 2 minutes to reject the duplicate as cross-post (ms)
+  cleanerInterval: 10 * 60 * 1000, // Frees up the memory from 10 to 10 minutes (ms)
+
+  lastMessageOfMember: {},
+
+  deleteMsgAndWarnUser: (msg, lastMsg) => {
+    msg.delete();
+    // prettier-ignore
+    msg.author
+        .send(`Hello there! It looks like you just posted a message to the #${msg.channel.name} channel on our server.
+			
+We don't allow cross-posting, so I had to delete your message because it is identical as your last one posted on #${lastMsg.channel.name}. You can avoid this warning deleting the first message or waiting 2 minutes.
+
+Thank you :)
+
+:robot: This message was sent by a bot, please do not respond to it - in case of additional questions / issues, please contact one of our mods!`);
+  },
+
+  handleMessage: ({ msg, user }) => {
+    if (deduper.monitoredChannels.includes(msg.channel.id)) {
+      const userId = user.id;
+      const lastMessage = deduper.lastMessageOfMember[userId];
+
+      if (
+        lastMessage &&
+        !lastMessage.deleted &&
+        lastMessage.channel.id !== msg.channel.id &&
+        lastMessage.content === msg.content &&
+        msg.createdTimestamp - lastMessage.createdTimestamp <
+          deduper.dupeExpireTime
+      ) {
+        deduper.deleteMsgAndWarnUser(msg, lastMessage);
+      } else {
+        deduper.lastMessageOfMember[userId] = msg;
```

Comment:
I like that this doesn't accumulate tons of messages, but it breaks in the case if they send a second message before cross posting. Maybe not super common, but would be nice to track user + channel, I think.

Reply:
Yeah currently it only sees the last message. The problem with user + channel is that the user could have sent a msg "X" in a channel and 10 days later, without sending other msg in this channel but in others, he could send "X" again in a different channel and it would be considerate as a cross-post.

Reply:
Wouldn't the
```
 msg.createdTimestamp - lastMessage.createdTimestamp <
          deduper.dupeExpireTime
```
above prevent that?

Reply:
Ah, right. I was thinking about another comment when I wrote this reply.